### PR TITLE
DCS-687 Deploying typescript app

### PR DESCRIPTION
**Pull request by andrewrlee**
Emitting compiled typescript and using generated source in docker image

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,34 +1,63 @@
-FROM node:12-buster-slim
```
Comment (andrewrlee) [inline Dockerfile:1]: have a story to bump to docker 14

File: backend/setupStaticContent.js [hunk 2]
```diff
@@ -4,15 +4,15 @@ const path = require('path')
 const router = express.Router()
 
 module.exports = () => {
-  router.use(express.static(path.join(__dirname, '../build')))
+  router.use(express.static(path.join(process.cwd(), '/build')))
```
Comment (andrewrlee) [inline backend/setupStaticContent.js:7]: make resilient to whether running inside `/dist` or `/app`

File: backend/tokenRefresher.js [hunk 3]
```diff
@@ -11,6 +11,7 @@ const logger = require('./log')
  * @param timeSinceTheEpochInSeconds
  */
 const tokenExpiresBefore = (encodedToken, timeSinceTheEpochInSeconds) => {
+  // @ts-ignore
   const token = jwtDecode(encodedToken)
```
Comment (andrewrlee) [inline backend/tokenRefresher.js:15]: type info doesn't work when using `require`, this should be fixed when we change to use `import`

**Review by JamieDraperUK — APPROVED** [review]
LGTM

File: tsconfig.json [hunk 4]
```diff
@@ -8,13 +8,13 @@
...
-    "noEmit": true,
+    "noEmit": false,
     "allowJs": true,
     "checkJs": true,
     "resolveJsonModule": true,
     "esModuleInterop": true,
     "allowSyntheticDefaultImports": true
   },
-  "exclude": ["node_modules", "assets/**/*.js", "integration-tests", "build", "static", "config",],
+  "exclude": ["node_modules", "assets/**/*.js", "integration-tests", "build", "dist/**/*.js", "static", "config",],
```
Comment (andrewrlee) [inline tsconfig.json:18]: this can be simplified to `["node_modules", "assets", "integration-tests", "build", "dist", "static", "config"]`

**Pull request merged**